src/main/java/com/mraof/minestuck/item/armor/JetPackItem.java

```diff
@@ -0,0 +1,217 @@
+package com.mraof.minestuck.item.armor;
+
+import com.mojang.logging.LogUtils;
+import com.mraof.minestuck.client.renderer.armor.JetpackModelRenderer;
+import com.mraof.minestuck.item.MSItems;
+import com.mraof.minestuck.util.MSParticleType;
+import com.mraof.minestuck.util.MSSoundEvents;
+import net.minecraft.client.model.HumanoidModel;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResultHolder;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.*;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.client.extensions.common.IClientItemExtensions;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import software.bernie.geckolib.animatable.GeoItem;
+import software.bernie.geckolib.animatable.SingletonGeoAnimatable;
+import software.bernie.geckolib.core.animatable.instance.AnimatableInstanceCache;
+import software.bernie.geckolib.core.animation.*;
+import software.bernie.geckolib.core.object.PlayState;
+import software.bernie.geckolib.util.GeckoLibUtil;
+
+import java.util.function.Consumer;
+
+public class JetPackItem extends ArmorItem implements GeoItem
+{
+	private final AnimatableInstanceCache cache = GeckoLibUtil.createInstanceCache(this);
+	private static final RawAnimation IDLE = RawAnimation.begin().thenPlay("jetpack.idle");
+	private static final RawAnimation PRIMED = RawAnimation.begin().thenPlay("jetpack.primed");
+	private static final RawAnimation FLIGHT = RawAnimation.begin().thenPlay("jetpack.flight");
+	private int EXPLOSION_BUFFER = 0;
+	
+	public static final Logger LOGGER = LogUtils.getLogger();
+	
+	public JetPackItem(ArmorMaterial mat, ArmorItem.Type slot, Properties props)
+	{
+		super(mat, slot, props);
+		
+		SingletonGeoAnimatable.registerSyncedAnimatable(this);
+	}
+	
+	@Override
+	public @NotNull InteractionResultHolder<ItemStack> use(@NotNull Level level, Player player, @NotNull InteractionHand hand)
+	{
+		ItemStack thrustController = new ItemStack(MSItems.THRUST_CONTROLLER.get());
+		
+		if(player.getInventory().getFreeSlot() <= 2)
+		{
+			player.getInventory().add(thrustController);
+		}
```

Comment:
no, i'm open to suggestions on a better way to go about this.

Reply:
Some options I can think of
- Crafting/alchemizing the thrust controllers
- Giving the jetpack a psuedo-storage nbt value to keep track of stored controllers. Similar to how SBURB computer block entities have a value to keep track of blank discs. When the jetpack is first made it will be able to give the player controllers but if the players lose track of said controllers it wont just automatically create new ones unless theyve been stowed back inside the jetpack